src/player_activity.h

```diff
@@ -59,39 +60,42 @@ class player_activity : public JsonSerializer, public JsonDeserializer
 {
     private:
         void finish( player *p );
-    public:
-        /** The type of this activity. */
-        activity_type type;
-        /** Total number of moves required to complete the activity */
-        int moves_total;
-        /** The number of moves remaining in this activity before it is complete. */
-        int moves_left;
-        /** An activity specific value. */
-        int index;
-        /** An activity specific value. */
-        int position;
-        /** An activity specific value. */
-        std::string name;
-        bool ignore_trivial;
-        std::vector<int> values;
-        std::vector<std::string> str_values;
-        std::vector<tripoint> coords;
-        tripoint placement;
-        /** If true, the player has been warned of dangerously close monsters with
-         * respect to this activity.
-         */
-        bool warned_of_proximity;
-        /** If true, the activity will be auto-resumed next time the player attempts
-         *  an identical activity. This value is set dynamically.
-         */
-        bool auto_resume;
 
-        player_activity( activity_type t = ACT_NULL, int turns = 0, int Index = -1, int pos = INT_MIN,
-                         std::string name_in = "" );
+    public:
+        player_activity() = default;
+        player_activity( const player_activity & );
+        player_activity &operator= ( const player_activity & );
         player_activity( player_activity && ) = default;
-        player_activity( const player_activity & ) = default;
         player_activity &operator=( player_activity && ) = default;
-        player_activity &operator=( const player_activity & ) = default;
+
+        player_activity( activity_type type, item_location &&target, int moves,
+                         std::string key = std::string(), const std::vector<int> &vals = {} )
+            : type( type ), moves_total( moves ), moves_left( moves ),
+              targets( {
+            {
+                std::move( target ), item_location(), item_location()
+            }
+        } ),
+        name( key ), values( vals ) {}
+
+        bool operator==( const player_activity &rhs ) const {
```

Comment:
Other code compares the internal members directly (for example in `player.cpp`) when checking if two activities are equivalent. This is brittle and even more so now I've added more members. Providing an equality operator prevents future breakage of that code if the class is further reorganised.